ql/src/java/org/apache/hadoop/hive/ql/hooks/HookContext.java

```diff
@@ -45,7 +47,50 @@
 public class HookContext {
 
   static public enum HookType {
-    PRE_EXEC_HOOK, POST_EXEC_HOOK, ON_FAILURE_HOOK
+
```

Comment:
I like this approach - could you make a small check:

* if we have hook compiled for the old api (which uses say the enum key `HookType.PRE_EXEC_HOOK`)
* will it work or not  (without recompilation) with the new implementation

Reply:
Checked on my test and production env,  it shows that the hooks compiled for the old api can be reused without any changes  with the new implementation.